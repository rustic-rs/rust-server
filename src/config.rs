pub mod auth_config;
pub mod auth_file;
pub mod server_config;
