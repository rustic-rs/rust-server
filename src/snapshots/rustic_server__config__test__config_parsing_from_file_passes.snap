---
source: src/config.rs
expression: config
---
RusticServerConfig {
    server: ConnectionSettings {
        listen: Some(
            127.0.0.1:8080,
        ),
    },
    storage: StorageSettings {
        data_dir: Some(
            "tests/generated/test_storage/",
        ),
        quota: None,
    },
    auth: HtpasswdSettings {
        disable_auth: false,
        htpasswd_file: Some(
            "tests/fixtures/test_data/.htpasswd",
        ),
    },
    acl: AclSettings {
        disable_acl: false,
        private_repos: true,
        append_only: false,
        acl_path: Some(
            "tests/fixtures/test_data/acl.toml",
        ),
    },
    tls: TlsSettings {
        disable_tls: false,
        tls_key: Some(
            "tests/fixtures/test_data/certs/test.key",
        ),
        tls_cert: Some(
            "tests/fixtures/test_data/certs/test.crt",
        ),
    },
    log: LogSettings {
        log_level: None,
        log_file: None,
    },
}
